src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> node;
+    private int size;
+    private Node<T> head = null;
+    private Node<T> tail = null;
+
     @Override
     public void add(T value) {
+        node = new Node<>(null, value,null);
+        if (head == null) {
+            head = tail = node;
+            size++;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            node.next = null;
+            tail = node;
+            size++;
+        }
```

Comment:
The `add(T value)` method can be simplified by removing the `else` block. You can create a new node and assign it to `tail.next` and `tail` regardless of whether the list is empty or not. If the list is empty (`head == null`), you should also assign the new node to `head`. Remember to increment `size` in both cases.